### ISSUE TITLE
Last trace always has no data leading to error in norm determination in linstack = stacking.linstack(traces)

### ISSUE THREAD

**Issue by caro-github** [eqcorrscan/EQcorrscan#74]
Here is the error message always for last trace:
```python
linstack = stacking.linstack(traces)
  File "/usr/local/lib/python2.7/dist-packages/eqcorrscan/utils/stacking.py", line 63, in linstack
    tr.data = np.sum((norm, tr.data), axis=0)
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py", line 1844, in sum
    out=out, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py", line 32, in _sum
    return umr_sum(a, axis, dtype, out, keepdims)
ValueError: operands could not be broadcast together with shapes (0,) (180000,) 
```

Here is my screen output with many print commands for more info:
```python
1 Trace(s) in Stream:
AF.WHYM.03.SHZ | 2009-03-16T17:32:26.770000Z - 2009-03-16T17:32:26.770000Z | 200.0 Hz, 0 samples
0
matchtr []
/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py:70: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
norm []
norm_nantonum []
[] 180000
```

**Comment by calum-chamberlain** [comment]
What have you typed to get the output below?
Is the formatting I have applied correct (see [mastering markdown](https://guides.github.com/features/mastering-markdown/) for how to format markdown)?
Is `traces` a list of streams as required [here](http://eqcorrscan.readthedocs.io/en/develop/submodules/autogen/eqcorrscan.utils.stacking.linstack.html#eqcorrscan.utils.stacking.linstack)?

**Comment by calum-chamberlain** [comment]
1. I meant the print statements, how did you get this:
```python
1 Trace(s) in Stream:
AF.WHYM.03.SHZ | 2009-03-16T17:32:26.770000Z - 2009-03-16T17:32:26.770000Z | 200.0 Hz, 0 samples
0
matchtr []
/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py:70: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
norm []
norm_nantonum []
[] 180000
```
?
2. I edited your comment so that the code output is formatted, does this look correct to you (as in is it what you saw in your terminal), if not, please format it - in general a working (or failing in this case, but reproducible) example is most helpful.
3. Golden, do all traces of all streams have data in them?

**Comment by caro-github** [comment]
1. I added print commands after every line where the problem occurs:         
    for i in range(1, len(streams)):
        for tr in stack:
            tr.plot()
            matchtr = streams[i].select(station=tr.stats.station,
                                        channel=tr.stats.channel)
            print(matchtr)
            print(len(matchtr[0].data))
            if matchtr:
                # Normalize the data before stacking
                if normalize:
	            print('matchtr',matchtr[0].data)

                    norm = matchtr[0].data /\
                        np.sqrt(np.mean(np.square(matchtr[0].data)))
	            print('norm',norm)
                    norm = np.nan_to_num(norm)
	            print('norm_nantonum',norm)
                else:
                    norm = matchtr[0].data
                print(norm,len(tr.data))
                tr.data = np.sum((norm, tr.data), axis=0)
    return stack

3. tr.plot() in 3.rd line shows me the data, so I don't know if it is the streams[i].select part or something else, if I change the order of the streams or remove the last stream, it's the same problem with whatever stream is last in the list, output for second last from same print commands is:
1 Trace(s) in Stream:
AF.WHYM.03.SHZ | 2009-03-15T04:59:58.295000Z - 2009-03-15T05:14:58.290000Z | 200.0 Hz, 180000 samples
180000
matchtr [ 0.00777793  0.05530805  0.18358414 ..., -1.03355332 -1.84595387
  0.00971802]
norm [ 0.00015242  0.00108381  0.00359751 ..., -0.02025346 -0.03617322
  0.00019043]
norm_nantonum [ 0.00015242  0.00108381  0.00359751 ..., -0.02025346 -0.03617322
  0.00019043]
[ 0.00015242  0.00108381  0.00359751 ..., -0.02025346 -0.03617322
  0.00019043] 180000

**Comment by calum-chamberlain** [comment]
It looks like you have a stream with an empty trace in it?
Please try to format code, it makes it much easier to read, it's really easy to do...

**Comment by caro-github** [comment]
I get the same problem with 2 other completely different datasets, and I already spent many hours on this. It is not as simple as one empty trace, I can assure you that

**Comment by calum-chamberlain** [comment]
Can you email me the streams?

**Comment by caro-github** [comment]
The streams are not empty but there is something weird about them, I think the first one might be a collection of traces rather than individual traces

**Comment by caro-github** [comment]
I have now been able to narrow down the problem with the data in traces disappearing: In ```if realign:``` in plotting.multi_trace_plot the line ```_pick.time -= shifts[i]``` resulted in an error because shifts[i] is a tuple and not a float, so I changed it to ```_pick.time -= shifts[i][1]``` which solved the problem but may have caused my problem. If you add the line ```print("Len Traces data", [len(traces[i].data) for i in range(0,len(traces))])``` before and after the if loop it results in the first value for the traces[0].data to be empty (but the stream not being empty because the header info is there) resulting in the problem in linstack. So I may have caused the problem myself through the correction I made, but I don't understand why this hadn't been an issue before.
